std/experimental/ndslice/slice.d

```diff
@@ -1178,6 +1178,29 @@ struct Slice(size_t _N, _Range)
             _ptr._range = range;
     }
 
+    /++
+    Returns:
+        pointer to the first element of a slice if slice is defined as `Slice!(N, T*)` or
+        plain structure with two fields `shift` and `range` otherwise. In second case the expression `range[shift]`
+        refers to the first element. For slices with named elements the type of a return value has the same behavior like a pointer.
+    Note:
+        `ptr` is defined only for not packed slices.
+    Attention:
+        `ptr` refers to the first element in the memory representation if and only if all strides are positive.
+    +/
+    static if (is(PureRange == Range))
```

Comment:
It would be helpful to add an else statement to this for an error message. So, instead of this:
```
Error: Slice does not have method ptr
```
You would get something like "Error: Cannot use ptr on a range..." etc.